x/genutil/src/balances_iter.rs

```diff
@@ -0,0 +1,56 @@
+use std::{collections::HashMap, path::Path};
+
+use gears::{
+    store::StoreKey,
+    types::{address::AccAddress, base::coins::UnsignedCoins},
+};
+
+#[derive(Debug, Clone, serde::Deserialize)]
+pub struct GenesisBalance {
+    pub address: AccAddress,
+    pub coins: UnsignedCoins,
+}
+
+#[derive(Debug, Clone)]
+pub struct GenesisBalanceIter(HashMap<AccAddress, UnsignedCoins>);
+
+impl GenesisBalanceIter {
+    pub fn new<SK: StoreKey>(sk: &SK, genesis_path: impl AsRef<Path>) -> anyhow::Result<Self> {
+        let mut value: serde_json::Value = serde_json::from_slice(&std::fs::read(genesis_path)?)?;
+
+        let value = value
+            .get_mut("app_state")
+            .ok_or(anyhow::anyhow!("missing `app_state`"))?
+            .get_mut(sk.name())
+            .ok_or(anyhow::anyhow!("module is not found"))?
+            .get_mut("balances")
```

Comment:
I'm slightly worried about this part `.get_mut("balances")`

Reply:
Why does genutils care about bank balances?

Reply:
'cause it needs to read balances of delegator. [Usage](https://github.com/NYBACHOK/gears/blob/6c8074953be648fa1d85a869cb761f9fc8c11f10/x/genutil/src/collect_txs.rs#L131-L141)

Reply:
OK